website/docs/aiml/chatbot/mistral.md

```diff
@@ -1,23 +1,100 @@
+[vLLM](https://github.com/vllm-project/vllm) is an open-source inference and serving engine specifically designed to optimize the performance of large language models (LLMs) through efficient memory management. As a popular inference solution in the ML community, vLLM offers several key advantages:
 
-In this lab serving the Mistral-7B-Instruct-v0.3 model is implemented using FastAPI, Ray Serve, and PyTorch-based Hugging Face Transformers to create an API for text generation.
+- **Efficient Memory Management**: Uses PagedAttention technology to optimize GPU/accelerator memory usage
+- **High Throughput**: Enables concurrent processing of multiple requests
+- **AWS Neuron Support**: Native integration with AWS Inferentia and Trainium accelerators
+- **OpenAI-compatible API**: Provides a drop-in replacement for OpenAI's API, simplifying integration
 
-Here's the code for compiling the model that we'll use:
+For AWS Neuron specifically, vLLM provides:
 
-```file
-manifests/modules/aiml/chatbot/ray-service-neuron-mistral-chatbot/mistral1.py
+- Native support for Neuron SDK and runtime
+- Optimized memory management for Inferentia/Trainium architectures
+- Continuous model loading for efficient scaling
+- Integration with AWS Neuron profiling tools
+
+For this lab, we will use the [Mistral-7B-v0.3 model](https://mistral.ai/news/announcing-mistral-7b) compiled with the `neuronx-distributed-inference` framework. This model provides a good balance between capabilities and resource requirements, making it suitable for deployment on our Trainium-powered EKS cluster.
+
+To deploy the model, we'll use a standard Kubernetes Deployment that will use a vLLM-based container image to load the model and weights:
+
+::yaml{file="manifests/modules/aiml/chatbot/vllm.yaml"}
+
+Let's create the necessary resources:
+
+```bash
+$ kubectl create namespace vllm
+$ kubectl apply -f ~/environment/eks-workshop/modules/aiml/chatbot/vllm.yaml
 ```
 
-This Python code performs the following tasks:
+We can check the resources it created:
+
+```bash
+$ kubectl get deployment -n vllm
+NAME      READY   UP-TO-DATE   AVAILABLE   AGE
+mistral   0/1     1            0           33s
+$ kubectl get service -n vllm
+NAME      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+mistral   ClusterIP   172.16.149.89   <none>        8080/TCP   33m
+```
+
+The model initialization process takes several minutes to complete. The vLLM Pod will go through the following stages:
+
+1. Remain in a Pending state until Karpenter provisions the Trainium instance
+2. Use an init container to download the model from Hugging Face to a host file system path
```

Comment:
should we explain that this is not a best practice because it requires download every time the pod is restarted?

Reply:
So because its being downloaded to the host file system path, the HF CLI is actually smart enough to skip downloading if its already there. Subsequent starts are much faster.